sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -86,7 +86,7 @@ internal static HttpMessageSanitizer CreateMessageSanitizer(DiagnosticsOptions d
 
         public async ValueTask<RequestFailedException> CreateRequestFailedExceptionAsync(Response response, ResponseError? error = null, IDictionary<string, string>? additionalInfo = null, Exception? innerException = null)
         {
-            if (GetType() == typeof(ClientDiagnostics) && error is null && additionalInfo is null)
+            if (GetType() == typeof(ClientDiagnostics) && response.ContentStream is null && error is null && additionalInfo is null)
```

Comment:
@AlexanderSher, it looks like you had added this check in https://github.com/Azure/azure-sdk-for-net/pull/33001, so I want to make sure I'm not undoing what you intended here.

I'd like to add this check for response.ContentStream == null to address the case that the client is using an unbuffered network stream.  If it is, the generator isn't passing error or additionalInfo, so we will call the RFE constructor, but it will fail when it tries to read the content from the network stream.  This check ensures that if there is content to read, it is handled properly by the existing method in ClientDiagnostics.